src/main/java/edu/ucsb/courses/controllers/ReactController.java

```diff
@@ -5,8 +5,8 @@
 
 @Controller
 public class ReactController {
-  @RequestMapping(value = {"/", "/{x:[\\w\\-]+}", "/{x:^(?!api$).*$}/**/{y:[\\w\\-]+}"})
-  public String getIndex() {
-    return "/index.html";
-  }
+   @RequestMapping(value = {"/", "/{x:^(?!swagger-ui$)[\\w\\-]+}", "/{x:^(?!api$).*$}/**/{y:[\\w\\-]+}"})
```

Comment:
Would it be possible to pull out these regexes into separate variables with some kind of self-documenting name, for better readability?

Reply:
That would be possible if I understood them.    

These regexes are inherited from code that Scott wrote, and I'm not sure I do understand them.  I know that the intent is to redirect everything that isn't an api call or a call to swagger-ui towards the React controller.    As for the details of why they are constructed the way they are... it's a mystery.    My concern is that if I try to get clever and refactor, I'll just break something.

 I did, however, make one change that might, at least, make that change I added a bit more self-explanatory.

Reply:
Unfortunately, my change to try to make it more readable broke it.  I think these regexes are best left alone when they work.   I'm putting it back to the way it was.

Reply:
I acknowledge that these regexes are difficult to read and understand.  But this is, honestly, the best I can do.    I think if we want to try to improve this, we may want to make that a separate issue. 

I really want to get this in so that I can teach from it next week if possible.